pkg/cmd/deploy/upload.go

```
@@ -1,136 +1,282 @@
+	bucketName string, progressBar *progressbar.ProgressBar) {
+	defer wg.Done()
+
+	var (
+		currentSize int64
+		filesToZip  []string
+	)
+
+	zipFileName := fmt.Sprintf("project_part_%d.zip", workerID)
 
-		// Skip node_modules and .edge directories
-		if info.IsDir() && (strings.Contains(path, "node_modules") || strings.Contains(path, ".edge")) {
-			logger.Debug("Skipping directory: " + path)
-			return filepath.SkipDir
+	for file := range filesChan {
+		filesToZip = append(filesToZip, file)
+
+		// Get the file size and add it to the current size
+		fileInfo, err := os.Stat(file)
+		if err != nil {
+			log.Printf("Failed to get file info for %s: %v", file, err)
+			continue
 		}
+		currentSize += fileInfo.Size()
 
-		if !info.IsDir() {
-			fileContent, err := cmd.Open(path)
+		// If the current size exceeds the limit (1MB), zip and upload
+		if currentSize >= maxZipSize {
+			err = zipAndUpload(ctx, cfg, filesToZip, zipFileName, baseDir,
+				atomicCounter, bucketName, progressBar)
 			if err != nil {
-				logger.Debug("Error while trying to read file <"+path+"> about to be uploaded", zap.Error(err))
-				return err
+				log.Printf("Failed to zip and upload files: %v", err)
 			}
 
-			fileString := strings.TrimPrefix(path, pathStatic)
-			mimeType, err := mimemagic.MatchFilePath(path, -1)
-			if err != nil {
-				logger.Debug("Error while matching file path", zap.Error(err))
-				return err
-			}
-			fileOptions := contracts.FileOps{
-				Path:        fileString,
-				MimeType:    mimeType.MediaType(),
-				FileContent: fileContent,
-			}
+			// Reset the size counter and clear the file list
+			currentSize = 0
+			filesToZip = nil
+		}
+	}
 
-			Jobs <- fileOptions
+	// If there are still files at the end, zip them up and upload them
+	if len(filesToZip) > 0 {
+		err := zipAndUpload(ctx, cfg, filesToZip, zipFileName, baseDir, atomicCounter, bucketName, progressBar)
+		if err != nil {
+			log.Printf("Failed to zip and upload files: %v", err)
```

Change this to a variable and return the error variable
Also: you SHOULD return the error, to stop the execution :)